cli/tests/integration_test.rs

```diff
@@ -47,13 +48,67 @@ mod tests {
             .assert();
         let bytes = assertion.get_output().stdout.as_slice();
         let output = str::from_utf8(bytes)?;
+        assert!(output.contains("-p"));
         assert!(output.contains("--port"));
         assert!(output.contains("The port to run [default: 3000]"));
+        assert!(output.contains("-v"));
         assert!(output.contains("--verbose"));
         assert!(output.contains("Verbose output"));
+        assert!(output.contains("-d"));
+        assert!(output.contains("--detach"));
+        assert!(output.contains("Run in detach mode"));
         Ok(())
     }
 
+    // @todo #129:35min Find a way to run slow tests separately from fast tests.
+    //  This test `accepts_request_in_detached_mode` runs a way longer than
+    //  other unit tests. Let's mark such long tests as slow and run them
+    //  separately from fast test. Locally, developers will run only fast
+    //  tests, while CI server will run both. Check
+    //  <a href="https://www.yegor256.com/2023/08/22/fast-vs-deep-testing.html">this link<a>
+    //  for more information about this idea.
+    #[tokio::test]
+    #[cfg(not(target_os = "windows"))]
+    // @todo #129:60min Create similar integration test for windows platform.
+    //  Now we have test for linux and macos. However, we need to maintain
+    //  similar test case for windows as well.
+    async fn accepts_request_in_detached_mode() -> Result<()> {
+        let assertion = Command::cargo_bin("cli")?.arg("start").arg("-d").assert();
+        let bytes = assertion.get_output().stdout.as_slice();
+        let output = str::from_utf8(bytes)?;
+        assert!(
+            output.contains("Server is running in detached mode on port 3000"),
+            "Output should contain logs that server started in detached mode"
+        );
+        let request = reqwest::Client::new();
+        let mut retries = 10;
+        let mut status = None;
+        while retries > 0 {
+            tokio::time::sleep(Duration::from_secs(1)).await;
+            match request.get("http://localhost:3000").send().await {
+                Ok(home) => {
+                    status = Some(home.status());
+                    break;
+                }
+                Err(_) => {
+                    retries -= 1;
+                }
+            }
+        }
+        assert_eq!(status.expect("Failed to retrieve status"), 200);
+        kill(3000);
```

Comment:
@h1alexbel `3000` can be moved to variable and used as parameter for creation. Also, I'm not sure that `kill` will be executed in 100% scenarios. Is it possible to use Java analog of try-with-resources?

Reply:
@h1alexbel I mean, during the test, if something goes wrong before the line `100` and the process stays alive after the test is completed, wdyt?

Reply:
@l3r8yJ I will add a puzzle for this

Reply:
@l3r8yJ what do you suggest for this? Something like destructor: https://www.yegor256.com/2017/08/08/raii-in-java.html?

Reply:
@h1alexbel i think we can use [defer](https://docs.rs/defer/latest/defer/). I think that you know how `defer` works

all that you need is to do this:

```rust
// start of the test
let _defer = defer(|| {
    kill(port)
});
// test body
```

Reply:
@l3r8yJ didn't know about it. Thanks